src/components/common/TableComponent.vue

```diff
@@ -1,11 +1,128 @@
 <template>
-  <div>Table Component</div>
+  <div class="table-wrapper">
+    <table class="table-container">
+      <thead>
+        <tr>
+          <th
+            v-for="(column, index) in columns"
+            :key="index"
+            @click="sortColumn(column.key)"
+            :data-sort="sortKey === column.key ? sortOrder : ''"
+          >
+            {{ column.label }}
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="(row, index) in sortedData" :key="index" class="table-data">
+          <td v-for="(column, index) in columns" :key="index">
+            {{ row[column.key] }}
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
 </template>
 
 <script>
 export default {
-  name: 'TableComponent',
+  props: {
+    columns: {
+      type: Array,
+      required: true,
+    },
+    data: {
+      type: Array,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      sortKey: '',
+      sortOrder: '',
+    };
+  },
+  computed: {
+    sortedData() {
```

Comment:
We can update the function.

    sortedData() {
      if (!this.sortKey) return this.data;
      return [...this.data].sort((a, b) => {
        const comparison = a[this.sortKey] > b[this.sortKey] ? 1 : -1;
        return this.sortOrder === 'asc' ? comparison : -comparison;
      });
    }